### PR TITLE
Title component with nivo ResponsiveWrapper

### LINKED ISSUE #107: use Nivo ResponsiveWrapper
thread for context: https://eqworks.slack.com/archives/GV827LS0M/p1598023495080100?thread_ts=1597787852.027300&cid=GV827LS0M

We use ChartWrapper to get access to the `height` and `width` as well as include a title in a styled div. Potentially could just switch from using `Responsive_____` to the "base" chart and manually include `ResponsiveWrapper`

### PR DESCRIPTION
**Pull request by geoerika**
Closes #107 #111

### CODE REVIEW

File: src/components/chart-wrapper/index.js [hunk 1]
```diff
@@ -32,25 +29,26 @@ const ChartInner = styled('div')`
...
-            </ChartInner>
-          )}
-        </AutoSizer>
-      </ChartContainer>
+      <ResponsiveWrapper>
+        {/* <AutoSizer> */}
+        {({ height, width }) => (
+          // THIS "height - 16" fixes bottom margin, but don't understand know how it works
+          // whithout it the bottom margin doesn't seem to adjust
+          <ChartInner height={height - 16} width={width}>
```
Comment (geoerika) [inline src/components/chart-wrapper/index.js:39]: This is the weird solution to have the bottom margin adjusting... can set it up maybe to 8...

Comment (Goondrious) [reply src/components/chart-wrapper/index.js:39]: I think this is a strange side effect of a few things:
- wrapping the chart with the `ChartInner` when `ResponsiveWrapper` already wraps in a div
- wrapping the underlying `ResponsiveBar/Scatter/Line/Pie` in another `ResponsiveWrapper`
- the `Title` div taking up extra space, causing the `ResponsiveWrapper` `{ height: 100% }` to overflow
- the `Wrapper` flex styling being at odds with ResizeObserver and/or react-grid-layout

Some weird behaviours:
- changing the value of `height - X` doesn't have much of an effect on the end result. It does seem to mess with how the animations look when **resizing**. e.g. change to `- 5` and it looks the same
- if you remove the `- X` and then tinker with the value of `h` in: https://github.com/EQWorks/chart-system/blob/9ccce2ee57091088cc39cf5d0becd2256691361b/stories/responsive-chart-wrapper.js#L20
  then you can see that the chart only looks messed up when you resize it. For example, keep it at `h: 6` and resize to ~h = 4, you'll see the chart extend past the container. Now change to `h: 4`and reload. The chart will look fine.

I think the following could be a good line:
- convert base charts from `ResponsiveX` to just `X` and then pass `width` and `height`
- remove `ChartInner`
- remove flex styling, which seemed to disrupt the resizing
- pass `height={height - adjustForTitle}`

did it here:

Comment (geoerika) [reply src/components/chart-wrapper/index.js:39]: Thanks Shane! I did try removing the ChartInner and have a non-responsive component, but didn't find your solution to be able to include the component... I will incorporate this now...

Comment (geoerika) [reply src/components/chart-wrapper/index.js:39]: The only problem we have to solve here, is when the title component upon shrinking wraps on the next line...

File: src/components/chart-wrapper/index.js [hunk 2]
```diff
@@ -18,23 +18,32 @@ const Title = styled('div', forwardRef)(
...
     margin: 0px 16px 8px 16px;
     overflow-wrap: anywhere;
-    color: ${ props['title-props'].color };
-    font-size: ${ props['title-props'].fontSize };
-    font-weight: ${ props['title-props'].fontWeight };
-    text-align: ${ props['title-props'].textAlign };
+    color: ${ props.color };
+    font-size: ${ props['font-size'] };
+    font-weight: ${ props['font-weight'] };
+    text-align: ${ props['text-align'] };
```
Comment (woozyking) [inline src/components/chart-wrapper/index.js:24]: how are undefined cases handled when one doesn't want to specify certain props? should we have safeguard values to fall back to?

File: src/shared/constants/title-props.js [hunk 3]
```diff
@@ -1,15 +1,34 @@
...
+
+// export const titleDefaultProps = {
+//   titleProps: {
+//     title: '',
+//     color: 'black',
+//     fontSize: '18px',
+//     fontWeight: 'normal',
+//     textAlign: 'left'
+//   }
+// }
```
Comment (woozyking) [inline src/shared/constants/title-props.js:34]: remove if unused

**Review by woozyking — APPROVED** [review]

File: src/components/chart-wrapper/index.js (hunk 2 above)
Comment (Goondrious) [reply src/components/chart-wrapper/index.js:24]: _preference:_ for such a straightforward relay to the style, unless these props are particularly important, I like `{ ...titleStyle }`. It's then just a catch-all. These are also less likely to change, so maybe even `style={titleStyle}`?

**Review by Goondrious — APPROVED** [review]

File: src/components/chart-wrapper/index.js (hunk 1 above)
Comment (Goondrious) [reply src/components/chart-wrapper/index.js]: getting the height from `ref` works!

Comment (geoerika) [reply src/components/chart-wrapper/index.js]: Yes, I used ref!

File: src/components/chart-wrapper/index.js (hunk 2 above)
Comment (geoerika) [reply src/components/chart-wrapper/index.js:24]: > how are undefined cases handled when one doesn't want to specify certain props? should we have safeguard values to fall back to?

Yes, I had default props defined. But, in light of what Shane proposed, which was more in line with what I wanted initially to do as well, I separated styling from title and, if the user wants to change the styling, we require that he specifies all styling.

**Review by woozyking — APPROVED** [review]
looks good but with conflicts

**Pull request merged**